dev/benchmarks/test_apps/stocks/lib/stocks_app.dart

```diff
@@ -0,0 +1,233 @@
+    return null;
+  }
+
+  @override
+  void dispose() {
+    _routerDelegate.dispose();
+    super.dispose();
+
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(() {
+      debugPaintSizeEnabled = _configuration.debugShowSizes;
+      debugPaintBaselinesEnabled = _configuration.debugShowBaselines;
+      debugPaintLayerBordersEnabled = _configuration.debugShowLayers;
+      debugPaintPointersEnabled = _configuration.debugShowPointers;
+      debugRepaintRainbowEnabled = _configuration.debugShowRainbow;
+      return true;
+    }());
+    return _StockConfigurationScope(
+      configuration: _configuration,
+      state: this,
+      child: MaterialApp.router(
+        title: 'Stocks',
+        theme: theme,
+        localizationsDelegates: StockStrings.localizationsDelegates,
+        supportedLocales: StockStrings.supportedLocales,
+        debugShowMaterialGrid: _configuration.debugShowGrid,
+        showPerformanceOverlay: _configuration.showPerformanceOverlay,
+        showSemanticsDebugger: _configuration.showSemanticsDebugger,
+        routeInformationParser: _routeInformationParser,
+        routerDelegate: _routerDelegate,
+      )
+    );
+  }
+}
+
+class _StockConfigurationScope extends InheritedWidget{
+  const _StockConfigurationScope({
+    Key key,
+    this.state,
+    this.configuration,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  final StockConfiguration configuration;
+  final StocksAppState state;
+
+  @override
+  bool updateShouldNotify(_StockConfigurationScope oldWidget) {
+    return oldWidget.configuration != configuration;
+  }
+}
+
+class StockRouteInformationParser extends RouteInformationParser<StockRoutePath> {
+  @override
+  Future<StockRoutePath> parseRouteInformation(RouteInformation routeInformation) {
+    final List<String> segments = routeInformation.routeName.split('?');
+    final String path = segments[0];
+    final String query = segments.length > 1 ? segments[1] : '';
+    final Uri syntheticUrl = Uri(path: path, query: query);
+    if (syntheticUrl.path == '/settings') {
+      return SynchronousFuture<StockRoutePath>(const StockSettingsPath());
+    } else {
+      if (syntheticUrl.path == '/stock') {
+        final String symbol = syntheticUrl.queryParameters['symbol'];
+        if (symbol != null && symbol.isNotEmpty)
+          return SynchronousFuture<StockRoutePath>(StockSymbolPath(symbol));
+      }
+      return SynchronousFuture<StockRoutePath>(const StockHomePath());
+    }
+  }
+
+  @override
+  RouteInformation restoreRouteInformation(StockRoutePath configuration) {
+    if (configuration is StockSettingsPath)
+      return const RouteInformation(routeName: '/settings');
+    if (configuration is StockHomePath)
+      return const RouteInformation(routeName: '/');
+    if (configuration is StockSymbolPath)
+      return RouteInformation(routeName: '/stock?symbol=${configuration.symbol}');
+    return null;
```

Comment:
What does `null` mean? Should there be an assert because configuration is always one of the options?